tests/Unit/Support/Constraint/IsEquivalentCssTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Support\Constraint;
+
+use Pelago\Emogrifier\Tests\Support\Constraint\IsEquivalentCss;
+use Pelago\Emogrifier\Tests\Support\Traits\CssDataProviders;
+use Pelago\Emogrifier\Tests\Support\Traits\TestStringConstraint;
+use PHPUnit\Framework\Constraint\Constraint;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Pelago\Emogrifier\Tests\Support\Constraint\IsEquivalentCss
+ */
+final class IsEquivalentCssTest extends TestCase
+{
+    use CssDataProviders;
+    use TestStringConstraint;
+
+    /**
+     * @test
+     *
+     * @param string $css
+     * @param string $otherCss
+     *
+     * @dataProvider provideEquivalentCss
+     */
+    public function matchesEquivalentCss(string $css, string $otherCss): void
+    {
+        $subject = new IsEquivalentCss($css);
+
+        $result = $subject->evaluate($otherCss, '', true);
+
+        self::assertTrue($result);
+    }
+
+    /**
+     * @test
+     *
+     * @param string $css
+     * @param string $otherCss
+     *
+     * @dataProvider provideNonEquivalentCss
+     */
+    public function notMatchesNonEquivalentCss(string $css, string $otherCss): void
+    {
+        $subject = new IsEquivalentCss($css);
+
+        $result = $subject->evaluate($otherCss, '', true);
+
+        self::assertFalse($result);
+    }
+
+    /**
+     * @return array<string, array{0: string, 1: string}>
+     */
+    public function provideNonEquivalentCss(): array
+    {
+        $datasets = $this->provideCssNeedleNotFoundInHaystack() + $this->provideCssNeedleFoundInLargerHaystack();
+
+        $transposedDatasets = \array_map(
+            static function (array $dataset): array {
```

Comment:
Please let's have PHPDoc type annotations for this static function.

Reply:
Done.  I've not added type annotations for the other closure because it's parameter and return value are both strings, so a DocBlock would not provide any additional information.